src/main/java/com/learning/browsers/Browser.java

```diff
@@ -1,20 +1,22 @@
 package com.learning.browsers;
 
-import com.learning.configuration.PropertiesReader;
-import javax.annotation.PreDestroy;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.edge.EdgeDriver;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
 
-public class Browser {
+@Component
+public class Browser implements DisposableBean {
 
-  private static final String BROWSER_PROPERTIES_NAME = "browser";
-  private static final String PATH_TO_EDGE_DRIVER = "pathToEdgeDriver";
-  private final PropertiesReader propertiesReader;
+  private final String browserName;
+  private final String pathToEdgeDriver;
   private WebDriver webDriver;
 
-  public Browser(PropertiesReader propertiesReader) {
-    this.propertiesReader = propertiesReader;
+  public Browser(@Value("${pathToEdgeDriver}") String pathToEdgeDriver, @Value("${browser}") String browserName) {
```

Comment:
Autowire `pathToEdgeDriver` via setter instead. This will make it clear this property is optional.

Reply:
Sure. I should catch this one.